test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1

```diff
@@ -6,249 +6,227 @@ Add-TestDynamicType
-
-	$result[0].Size              | Should -Be ($orig1 + 1)
-	$dirObject[0].Size           | Should -Be ($orig1 + 1)
-	$dirObject[$TestLength].Size | Should -Be ($orig2 + 1)
-	}
-
-	It "Should not leak 'StopUpstreamCommandsException' internal exception for stopping upstream" {
-		1,2 | Select-Object -First 1 -ErrorVariable err
-		$err | Should -BeNullOrEmpty
-	}
-}
+        $orig1 = $dirObject[0].Size
+        $orig2 = $dirObject[$TestLength].Size
+        $result = $dirObject | addOneToSizeProperty | Select-Object -First $TestLength -Wait
 
-Describe "Select-Object DRT basic functionality" -Tags "CI" {
-	BeforeAll {
-		$employees = [pscustomobject]@{"FirstName"="joseph"; "LastName"="smith"; "YearsInMS"=15},
-                            [pscustomobject]@{"FirstName"="paul"; "LastName"="smith"; "YearsInMS"=15},
-                            [pscustomobject]@{"FirstName"="mary"; "LastName"="soe"; "YearsInMS"=5},
-                            [pscustomobject]@{"FirstName"="edmund"; "LastName"="bush"; "YearsInMS"=9}
-	}
-
-	It "Select-Object with empty script block property should throw"{
-		$e = { "bar" | Select-Object -Prop {} -ErrorAction Stop } |
-			Should -Throw -ErrorId "EmptyScriptBlockAndNoName,Microsoft.PowerShell.Commands.SelectObjectCommand" -PassThru
-		$e.CategoryInfo | Should -Match "PSArgumentException"
-	}
-
-	It "Select-Object with string property should work"{
-		$result = "bar" | Select-Object -Prop foo | Measure-Object
-		$result.Count | Should -Be 1
-	}
+        $result[0].Size              | Should -Be ($orig1 + 1)
+        $dirObject[0].Size           | Should -Be ($orig1 + 1)
+        $dirObject[$TestLength].Size | Should -Be ($orig2 + 1)
+    }
 
-	It "Select-Object with Property First Last Overlap should work"{
-		$results = $employees | Select-Object -Property "YearsInMS", "L*" -First 2 -Last 3
+    It "Should not leak 'StopUpstreamCommandsException' internal exception for stopping upstream" {
+        1, 2 | Select-Object -First 1 -ErrorVariable err
+        $err | Should -BeNullOrEmpty
+    }
+}
 
-		$results.Count | Should -Be 4
+Describe "Select-Object DRT basic functionality" -Tags "CI" {
+    BeforeAll {
+        $employees = @(
+            [pscustomobject]@{"FirstName" = "joseph"; "LastName" = "smith"; "YearsInMS" = 15 }
+            [pscustomobject]@{"FirstName" = "paul"; "LastName" = "smith"; "YearsInMS" = 15 }
+            [pscustomobject]@{"FirstName" = "mary"; "LastName" = "soe"; "YearsInMS" = 5 }
+            [pscustomobject]@{"FirstName" = "edmund"; "LastName" = "bush"; "YearsInMS" = 9 }
+        )
+    }
 
-		$results[0].LastName | Should -Be $employees[0].LastName
-		$results[1].LastName | Should -Be $employees[1].LastName
-		$results[2].LastName | Should -Be $employees[2].LastName
-		$results[3].LastName | Should -Be $employees[3].LastName
+    It "Select-Object with empty script block property should throw" {
+        $e = { "bar" | Select-Object -Prop { } -ErrorAction Stop } |
```

Comment:
This test failed in CI. It turns out `"bar" | Select-Object -Prop {} -ErrorAction Stop` throws, but `{ }` doesn't. It must be a bug :)

Reply:
Yeah, the check done in that particular case is a bit too simplistic, for sure. 😄

Reply:
The check is done here:

https://github.com/PowerShell/PowerShell/blob/f4b8ae5e048a5d2c2d8d06b36e4ac1c827346fdf/src/Microsoft.PowerShell.Commands.Utility/commands/utility/Select-Object.cs#L464-L495

It looks like it's mainly done this way to ensure properties with an empty string as the name can't be created. Technically a property name can be used if it's just a space, though it's not really discoverable... so I guess it might be intentional, though I question whether it's very useful functionality to enable that. 😂

Reply:
I'm fine keeping as is. Just update the test to use `{}` like before and that should be fine.